### PR TITLE
Dependency collector should not be http specific

### PR DESCRIPTION
**Pull request by lmolkova**
Today DependenciesCollector uses sampler factory like Func<HttpRequestMessage, ISampler>. The goal is to sample out all calls to internal endpoints.

This is not the right way to do it - the proper way is through DiagnosticSource.IsEnabled callback. 

If we use sampler for it, we should create builder, span and do a lot of things before we can discard call that we do want not track.

So this mechanism now requires every custom listener to work with Http message and it must change in future. 
I'm removing it now and we'll address url filtering in #166 using DiagnostiSource.IsEnabled.

### CODE REVIEW

File: src/OpenTelemetry.Collector.Dependencies/DependenciesCollector.cs [hunk 1]
```diff
@@ -35,28 +35,16 @@ public class DependenciesCollector : IDisposable
         /// </summary>
         /// <param name="options">Configuration options for dependencies collector.</param>
         /// <param name="tracer">Tracer to record traced with.</param>
-        /// <param name="sampler">Sampler to use to sample dependnecy calls.</param>
+        /// <param name="sampler">Sampler to use to sample dependency calls.</param>
         public DependenciesCollector(DependenciesCollectorOptions options, ITracer tracer, ISampler sampler)
```
Comment (pakrym) [inline src/OpenTelemetry.Collector.Dependencies/DependenciesCollector.cs]: Why have both sampler and custom sampler?

Comment (lmolkova) [reply src/OpenTelemetry.Collector.Dependencies/DependenciesCollector.cs]: good point. It seems there is no reason to have custom sampler now. And if someone wants it, they can use `DependenciesCollector` constructor to provide it. Will remove one from options

**Comment by pakrym** [comment]
LGTM

**Comment by lmolkova** [comment]
@austinlparker @discostu105 please review

**Comment by lmolkova** [comment]
@austinlparker @discostu105 ping?

File: src/OpenTelemetry.Collector.Dependencies/DependenciesCollector.cs (hunk 1 above)
Comment (SergeyKanzhelev) [reply src/OpenTelemetry.Collector.Dependencies/DependenciesCollector.cs]: sampler is globally registered one. CustomSampler is for filtering of a calls to Zipkin or such. CustomSampler falls back to one in constructor when doen't know what to do with the request. Also custom sampler works on raw http object so can sample based on url. BUT it is specific to the implementation

**Comment by lmolkova** [comment]
@SergeyKanzhelev 

This is not the right way to do it - the proper way is through DiagnosticSource.IsEnabled callback. And you see it checks for wrong URL anyway.

If we use sampler for it, we should create builder, span and do a lot of things before we can discard call that we do want not track.

So this mechanism now requires every custom listener to work with Http message and it must change in future.
I'm removing it now and we'll address url filtering in #166 using DiagnostiSource.IsEnabled.

**Comment by lmolkova** [comment]
Chatted with Sergey offline. Dropping this for now, will do it in the scope of #166 with a proper callback to filter urls

**Pull request closed without merging**